File: java/socketd/src/main/java/org/noear/socketd/transport/core/fragment/FragmentHandlerDefault.java
package org.noear.socketd.transport.core.fragment;

import org.noear.socketd.transport.core.*;

import java.io.IOException;

/**
 * 数据分片默认实现
 *
 * @author noear
 * @since 2.0
 */
public class FragmentHandlerDefault extends FragmentHandlerBase {
    /**
     * 创建分片聚合器
     */
    @Override
    protected FragmentAggregator createFragmentAggregator(MessageInternal message) throws IOException {
        return new FragmentAggregatorDefault(message);
    }

    @Override
    public boolean aggrEnable() {
        return true;
    }
}

File: python/socketd/socketd/transport/core/fragment/FragmentHandlerDefault.py

from socketd.transport.core.Message import MessageInternal
from socketd.transport.core.FragmentAggregator import FragmentAggregator

from .FragmentAggregatorDefault import FragmentAggregatorDefault
from .FragmentHandlerBase import FragmentHandlerBase


class FragmentHandlerDefault(FragmentHandlerBase):

    def create_fragment_aggregator(self, message: MessageInternal) -> FragmentAggregator:
        return FragmentAggregatorDefault(message)

    def aggr_enable(self) -> bool:
        return True
